chore: cleanup outdated services and old links

lidel opened the pull request:
This PR includes some automated cleanup of outdated tutorials:

Updated pinning service documentation:
- Removed outdated web3.storage references from quickstart guides
- Added current pinning services: Storacha, Pinata, and Filebase
- Split `/quickstart/publish.md` and `/quickstart/publish_cli.md` into distinct GUI-focused and CLI-focused guides
- Removed unused web3.storage screenshots

Fixed outdated references:
- Updated Accelerated DHT Client links from experimental features to current config documentation
- Removed unnecessary "go-ipfs" references (now just "Kubo")

Improved main install page:
- Added "Publish files with a pinning service" section for easy onboarding

Files modified
- `docs/quickstart/publish.md` - Focus on web/GUI interfaces
- `docs/quickstart/publish_cli.md` - Focus on CLI/API methods
- `docs/install/README.md` - Added pinning section, minor fixes
- `docs/how-to/gateway-best-practices.md` - Updated DHT client link
- `docs/how-to/troubleshooting.md` - Updated DHT client link

Files removed
- `docs/quickstart/images/` - Removed 6 unused web3.storage screenshots


## Checklist before requesting a review
- [ ] Passing the beta version of the **Check Markdown links for modified files** check. Action results can be viewed [here](https://github.com/ipfs/ipfs-docs/actions/workflows/action.yml).

## Checklist before merging
- [ ] Passing all required checks (The beta **Check Markdown links for modified files** check is not required)

2color commented on docs/quickstart/publish.md:
```diff
@@ -5,22 +5,18 @@ description: 'Learn how to publish content with IPFS by pinning a file to a pinn
 
 # Publish a file with IPFS
```
If we're bringing this page back I'd add it back to the sidebar menu

2color commented:
If we're bringing back the publish and publish cli pages, we should add it back to the sidebar menu.

2color reviewed (commented):
Nice. LGTM overall. I left a few small comments.

2color commented on docs/quickstart/publish_cli.md:
```diff
@@ -5,22 +5,18 @@ description: 'Learn how to publish content with IPFS by pinning a file to a pinn
 
 # Publish a file with IPFS using the command line
```
```suggestion
# Pin a file with IPFS using the command line
```

lidel commented:
@2color mind taking a final look? added to side menu, renamed them from "publish" to "pinning" and added cross-references with glossary and existing self-hosting tutorials for GUI and CLI.

2color commented on docs/.vuepress/config.js:
```diff
@@ -105,7 +105,9 @@ module.exports = {
               children: [
-                ['/quickstart/retrieve','Retrieve with IPFS'],
+                ['/quickstart/retrieve','Retrieve from IPFS'],
+                ['/quickstart/pin','Pin with Browser'],
```
```suggestion
                ['/quickstart/pin','Pin (Browser)'],
```

2color commented on docs/.vuepress/config.js:
```diff
@@ -105,7 +105,9 @@ module.exports = {
-                ['/quickstart/retrieve','Retrieve with IPFS'],
+                ['/quickstart/retrieve','Retrieve from IPFS'],
+                ['/quickstart/pin','Pin with Browser'],
+                ['/quickstart/pin-cli','Pin with Command Line'],
```
```suggestion
                ['/quickstart/pin-cli','Pin (CLI)'],
```

2color commented on docs/.vuepress/config.js:
```diff
@@ -123,7 +125,9 @@ module.exports = {
               children: [
-                ['/quickstart/retrieve','Retrieve with IPFS'],
+                ['/quickstart/retrieve','Retrieve from IPFS'],
+                ['/quickstart/pin','Pin with Browser'],
```
```suggestion
                ['/quickstart/pin','Pin (Browser)'],
```

2color commented on docs/.vuepress/config.js:
```diff
@@ -123,7 +125,9 @@ module.exports = {
-                ['/quickstart/retrieve','Retrieve with IPFS'],
+                ['/quickstart/retrieve','Retrieve from IPFS'],
+                ['/quickstart/pin','Pin with Browser'],
+                ['/quickstart/pin-cli','Pin with Command Line'],
```
```suggestion
                ['/quickstart/pin-cli','Pin (CLI)'],
```
